blockchain/block.go

```diff
@@ -8,6 +8,15 @@ import (
 	"io"
 )
 
+const (
```

Comment:
If we're going to use these constants, we should make tests that assert their size is correct so that if any extra fields get added to the block header for example, and the developer doesn't change the size constant, it will fail.

This can be done by creating a block header, and asserting that the len(blockheader.Marshal()) is the same as the BlockHeaderLen. (Assuming that the developer appended the extra fields in the marshalling function)

It might also be a good idea to make these read only by making them private and creating "getter" functions for them.